src/Neo/Extensions/SmartContract/NeoTokenExtensions.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// NeoTokenExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Neo.Extensions
+{
+    public static class NeoTokenExtensions
+    {
+        public static IEnumerable<(UInt160 Address, BigInteger Balance)> GetAccounts(this NeoToken neoToken, DataCache snapshot)
+        {
+            if (snapshot is null)
+                throw new ArgumentNullException(nameof(snapshot));
+
+            var kb = new KeyBuilder(neoToken.Id, NeoToken.Prefix_Account);
+            var prefixKey = kb.ToArray();
+
+            foreach (var (key, value) in snapshot.Seek(prefixKey, SeekDirection.Forward))
```

Comment:
`DataCache ` has a method: `Find(byte[] key_prefix = null, SeekDirection direction = SeekDirection.Forward)`, and it can get all items with the prefix, so maybe `Find(prefixKey)` can be used here?

Reply:
but i dont want it cached or as `StorageItem`

Reply:
> but i dont want as `StorageItem` and wont be able to get `GetInteroperable`

`Seek` and `Find` return `IEnumerable<(StorageKey Key, StorageItem Value)>` both.

Reply:
no they dont

Reply:
> no they dont

https://github.com/neo-project/neo/blob/ac8e532d6fd65e866c4313bdb4079d2b4e128226/src/Neo/Persistence/DataCache.cs#L427

https://github.com/neo-project/neo/blob/ac8e532d6fd65e866c4313bdb4079d2b4e128226/src/Neo/Persistence/DataCache.cs#L218

Reply:
For some reason I thought I was using `IStore`